lib/ansible/modules/remote_management/redfish/redfish.py

```diff
@@ -0,0 +1,331 @@
+    description:
+      - value of BIOS attribute to update to
+  mgr_attr_name:
+    required: false
+    description:
+      - name of Manager attribute to update
+  mgr_attr_value:
+    required: false
+    description:
+      - value of Manager attribute to update to
+
+author: "Jose Delarosa (github: jose-delarosa)"
+'''
+
+EXAMPLES = '''
+  - name: Getting system inventory
+    local_action: >
+       redfish category=Inventory command=GetSystemInventory baseuri={{baseuri}}
+       user={{user}} password={{password}}
+
+  - name: Get CPU Inventory
+    local_action: >
+       redfish category=Inventory command=GetCpuInventory baseuri={{baseuri}}
+       user={{user}} password={{password}}
+
+  - name: Enable PXE Boot for NIC1
+    local_action: >
+       redfish category=System command=SetBiosAttributes
+       user={{user}} password={{password}} baseuri={{baseuri}}
+       bios_attr_name=PxeDev1EnDis bios_attr_value=Enabled
+
+  - name: Set one-time boot device to {{bootdevice}}
+    local_action: >
+       redfish category=System command=SetOneTimeBoot baseuri={{baseuri}}
+       user={{user}} password={{password}} bootdevice={{bootdevice}}
+
+  - name: Turn system power on
+    local_action: >
+       redfish category=System command=PowerOn baseuri={{baseuri}}
+       user={{user}} password={{password}}
+
+  - name: Add user
+    local_action: >
+       redfish category=Accounts command=AddUser baseuri={{baseuri}}
+       user={{user}} password={{password}} userid={{userid}}
+       username={{username}} userpswd={{userpswd}} userrole={{userrole}}
+
+  - name: Enable user
+    local_action: >
+       redfish category=Accounts command=EnableUser baseuri={{baseuri}}
+       user={{user}} password={{password}} userid={{userid}}
+
+  - name: Get BIOS attributes
+    local_action: >
+       redfish category=System command=GetBiosAttributes baseuri={{baseuri}}
+       user={{user}} password={{password}}
+'''
+
+RETURN = '''
+changed:
+    description: Indicates if any changes were made
+    returned: always
+    type: boolean
+    sample: false
+invocation:
+    description: module arguments passed
+    returned: always
+    type: string
+    sample: module_args
+result:
+    description: different results depending on task
+    returned: always
+    type: string
+    sample: BIOS Attributes set as pending values
+'''
+
+import os
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.redfish_utils import RedfishUtils
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, type='str'),
+            command=dict(required=True, type='str'),
+            baseuri=dict(required=True, type='str'),
+            user=dict(required=True, type='str'),
+            password=dict(required=True, type='str', no_log=True),
+            userid=dict(required=False, type='str'),
+            username=dict(required=False, type='str'),
```

Comment:
`required=False` and `type='str'` are default, so no need to explicitly state them in argument_spec